File: cloud/acp-admin-cloud-dependencies/src/main/java/pers/acp/admin/common/po/FlowStartPO.java
package pers.acp.admin.common.po;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

import javax.validation.constraints.NotBlank;
import java.util.HashMap;
import java.util.Map;

/**
 * @author zhang by 14/06/2019
 * @since JDK 11
 */
@ApiModel("启动流程参数")
public class FlowStartPO {

    public String getProcessDefinitionKey() {
        return processDefinitionKey;
    }

    public void setProcessDefinitionKey(String processDefinitionKey) {
        this.processDefinitionKey = processDefinitionKey;
    }

    public String getBusinessKey() {
        return businessKey;
    }

    public void setBusinessKey(String businessKey) {
        this.businessKey = businessKey;
    }

    public Map<String, Object> getParams() {
        return params;
    }

    public void setParams(Map<String, Object> params) {
        this.params = params;
    }

    @ApiModelProperty(value = "流程定义键", required = true, position = 1)
    @NotBlank(message = "流程定义键不能为空")
    private String processDefinitionKey;

    @ApiModelProperty(value = "业务键", required = true, position = 2)
    @NotBlank(message = "业务键不能为空")
    private String businessKey;

    @ApiModelProperty(value = "自定义参数", position = 3)
    private Map<String, Object> params = new HashMap<>();

}

File: cloud/acp-admin-cloud-dependencies/src/main/kotlin/pers/acp/admin/common/po/FlowStartPo.kt
package pers.acp.admin.common.po

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

import javax.validation.constraints.NotBlank

/**
 * @author zhang by 14/06/2019
 * @since JDK 11
 */
@ApiModel("启动流程参数")
data class FlowStartPo(
        @ApiModelProperty(value = "流程定义键", required = true, position = 1)
        @get:NotBlank(message = "流程定义键不能为空")
        var processDefinitionKey: String? = null,

        @ApiModelProperty(value = "业务键", required = true, position = 2)
        @get:NotBlank(message = "业务键不能为空")
        var businessKey: String? = null,

        @ApiModelProperty(value = "自定义参数", position = 3)
        var params: MutableMap<String, Any> = mutableMapOf()
)
